src/services/completions.ts

```diff
@@ -675,6 +675,37 @@ namespace ts.Completions {
             hasAction = !importCompletionNode;
         }
 
+        const kind = SymbolDisplay.getSymbolKind(typeChecker, symbol, location); // TODO: GH#18217
+        if (kind === ScriptElementKind.jsxAttribute && preferences.jsxSnippetCompletion && preferences.jsxSnippetCompletion !== "none") {

```

Comment:
I think you should also check `preferences.includeCompletionsWithSnippetText` since that indicates editor support for snippets overall.

Reply:
Done.